src/Java.Interop.Tools.Generator/Extensions/XmlExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Xml.Linq;
+using System.Xml.XPath;
+
+namespace Xamarin.Android.Tools
+{
+	static class XmlExtensions
+	{
+		public static string? XGetAttribute (this XElement element, string name)
+			=> element.Attribute (name)?.Value.Trim ();
+
+		public static string? XGetAttribute (this XPathNavigator nav, string name, string ns)
+			=> nav.GetAttribute (name, ns)?.Trim ();
+
+		public static int? XGetAttributeAsInt (this XElement element, string name)
+		{
+			var value = element.XGetAttribute (name);
+
+			if (int.TryParse (value, out var result))
```

Comment:
One possible concern here is that this uses the current culture info to parse `value`: https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=net-5.0#System_Int32_TryParse_System_String_System_Int32__

> The `s` parameter is parsed using the formatting information in a [NumberFormatInfo](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.numberformatinfo?view=net-5.0) object initialized for the current system culture.

We may want to instead use:

```csharp
int.TryParse (value, NumberStyles.Integer, CurrentCulture.InvariantCulture, out var result)
```

Reply:
Alternate idea: why don't we use the [explicit conversion from `XAttribute` to `int?` operator](https://docs.microsoft.com/en-us/dotnet/api/system.xml.linq.xattribute.op_explicit?view=net-5.0#System_Xml_Linq_XAttribute_op_Explicit_System_Xml_Linq_XAttribute__System_Nullable_System_Int32_)?

```csharp
public static int? XGetInt32Attribute (this XElement element, string name) => (int?) element.Attribute (name);
```

Reply:
This seems to throw an exception on invalid input:
```
System.FormatException
  HResult=0x80131537
  Message=Input string was not in a correct format.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.Number.ParseInt32(ReadOnlySpan`1 value, NumberStyles styles, NumberFormatInfo info)
   at System.Int32.Parse(String s, NumberStyles style, IFormatProvider provider)
   at System.Xml.XmlConvert.ToInt32(String s)
   at System.Xml.Linq.XAttribute.op_Explicit(XAttribute attribute)
```